.github/workflows/release.yml

```diff
@@ -0,0 +1,43 @@
+name: release
+
+on:
+  push:
+    branches:
+      - master
+
+  workflow_dispatch:
+
+jobs:
+  pre_run:
+    name: Cancel previous runs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@ad6cb1b847ffb509a69b745b6ee2f1d14dfe14b8
+        with:
+          access_token: ${{ github.token }}
+
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+
+    if: "github.event_name == 'push' && contains(github.event.head_commit.message, 'chore: publish')"
+    needs:
+      - pre_run
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.GH_TOKEN }}
```

Comment:
do we need the `with` section here? @CharlieC3

Reply:
Yeah, it could break the release process without it. [Full explanation why here](https://github.com/hirosystems/gh-actions-example/blob/main/.github/workflows/ci.yml#L116-L118).